ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -138,7 +158,15 @@ - (void)transactionalComponentDataSource:(CKTransactionalComponentDataSource *)d
     CKComponentBoundsAnimationApply(boundsAnimation, ^{
       for (NSIndexPath *indexPath in changes.updatedIndexPaths) {
         CKTransactionalComponentDataSourceItem *item = [state objectAtIndexPath:indexPath];
+          
+        // There is a race condition that causes this method to return nil
+        // between when the cell is first requested and when the cell is
+        // placed in the collection view. Thus we fall back on our mapping.
```

Comment:
Great comment.

I'd like to understand the problem a bit more. Is this issue orthogonal to your optimization or is it something you noticed as a result of the optimization? Just trying to gauge whether or not I need to consider these changes together or if I can think about them separately.